### ISSUE TITLE
docx to latex: images and captions are not being converted properly

### ISSUE THREAD

**Issue by Ultrahero** [jgm/pandoc#6893]
Pandoc version 2.11.1.1
command: pandoc input_file -f docx -t latex -s -o output_file.tex

Hi,
I stumbled across another issue:
An image I copied into word and added a caption to were saved in the docx file,
but not properly converted to .tex. The caption was missing altogether and the image itself was not included only the reference.

[pandoc_image_error.docx](https://github.com/jgm/pandoc/files/5602984/pandoc_image_error.docx)
[image_error.tex.txt](https://github.com/jgm/pandoc/files/5602985/image_error.tex.txt)

**Comment by jgm** [comment]
I can look into the missing caption.
As for

> the image itself was not included only the reference

what you need is to use the `--extract-media=DIR` option.  Then the image will be extracted.  Pandoc won't litter your file system with files unless you explicitly tell it where they should go.

**Comment by jgm** [comment]
Here's the xml:
``` xml
    <w:p w14:paraId="0BCF1C3B" w14:textId="3880F9C2" w:rsidR="00043FF5" w:rsidRPr="007C1E2B" w:rsidRDefault="00A80F5E">
      <w:r w:rsidRPr="00A80F5E">
        <w:drawing>
          <wp:anchor distT="0" distB="0" distL="114300" distR="114300" simplePos="0" relativeHeight="251659264" behindDoc="0" locked="0" layoutInCell="1" allowOverlap="1" wp14:anchorId="67E57856" wp14:editId="681F0AC3">
            <wp:simplePos x="0" y="0"/>
            <wp:positionH relativeFrom="column">
              <wp:align>center</wp:align>
            </wp:positionH>
            <wp:positionV relativeFrom="page">
              <wp:posOffset>1184275</wp:posOffset>
            </wp:positionV>
            <wp:extent cx="2165350" cy="854075"/>
            <wp:effectExtent l="0" t="0" r="6350" b="3175"/>
            <wp:wrapTopAndBottom/>
            <wp:docPr id="1" name="Grafik 1"/>
            <wp:cNvGraphicFramePr>
              <a:graphicFrameLocks xmlns:a="http://schemas.openxmlformats.org/drawingml/2006/main" noChangeAspect="1"/>
            </wp:cNvGraphicFramePr>
            <a:graphic xmlns:a="http://schemas.openxmlformats.org/drawingml/2006/main">
              <a:graphicData uri="http://schemas.openxmlformats.org/drawingml/2006/picture">
                <pic:pic xmlns:pic="http://schemas.openxmlformats.org/drawingml/2006/picture">
                  <pic:nvPicPr>
                    <pic:cNvPr id="0" name="Picture 1"/>
                    <pic:cNvPicPr>
                      <a:picLocks noChangeAspect="1" noChangeArrowheads="1"/>
                    </pic:cNvPicPr>
                  </pic:nvPicPr>
                  <pic:blipFill>
                    <a:blip r:embed="rId4">
                      <a:extLst>
                        <a:ext uri="{28A0092B-C50C-407E-A947-70E740481C1C}">
                          <a14:useLocalDpi xmlns:a14="http://schemas.microsoft.com/office/drawing/2010/main" val="0"/>
                        </a:ext>
                      </a:extLst>
                    </a:blip>
                    <a:srcRect/>
                    <a:stretch>
                      <a:fillRect/>
                    </a:stretch>
                  </pic:blipFill>
                  <pic:spPr bwMode="auto">
                    <a:xfrm>
                      <a:off x="0" y="0"/>
                      <a:ext cx="2165350" cy="854075"/>
                    </a:xfrm>
                    <a:prstGeom prst="rect">
                      <a:avLst/>
                    </a:prstGeom>
                    <a:noFill/>
                    <a:ln>
                      <a:noFill/>
                    </a:ln>
                  </pic:spPr>
                </pic:pic>
              </a:graphicData>
            </a:graphic>
          </wp:anchor>
        </w:drawing>
      </w:r>
      <w:r w:rsidRPr="00A80F5E">
        <mc:AlternateContent>
          <mc:Choice Requires="wps">
            <w:drawing>
              <wp:anchor distT="0" distB="0" distL="114300" distR="114300" simplePos="0" relativeHeight="251660288" behindDoc="0" locked="0" layoutInCell="1" allowOverlap="1" wp14:anchorId="4AFB6FA3" wp14:editId="09ECCCE2">
                <wp:simplePos x="0" y="0"/>
                <wp:positionH relativeFrom="column">
                  <wp:align>center</wp:align>
                </wp:positionH>
                <wp:positionV relativeFrom="paragraph">
                  <wp:posOffset>1119505</wp:posOffset>
                </wp:positionV>
                <wp:extent cx="2518410" cy="635"/>
                <wp:effectExtent l="0" t="0" r="0" b="6985"/>
                <wp:wrapTopAndBottom/>
                <wp:docPr id="2" name="Textfeld 2"/>
                <wp:cNvGraphicFramePr/>
                <a:graphic xmlns:a="http://schemas.openxmlformats.org/drawingml/2006/main">
                  <a:graphicData uri="http://schemas.microsoft.com/office/word/2010/wordprocessingShape">
                    <wps:wsp>
                      <wps:cNvSpPr txBox="1"/>
                      <wps:spPr>
                        <a:xfrm>
                          <a:off x="0" y="0"/>
                          <a:ext cx="2518410" cy="635"/>
                        </a:xfrm>
                        <a:prstGeom prst="rect">
                          <a:avLst/>
                        </a:prstGeom>
                        <a:solidFill>
                          <a:prstClr val="white"/>
                        </a:solidFill>
                        <a:ln>
                          <a:noFill/>
                        </a:ln>
                      </wps:spPr>
                      <wps:txbx>
                        <w:txbxContent>
                          <w:p w14:paraId="6D60BF4E" w14:textId="77777777" w:rsidR="00A80F5E" w:rsidRPr="00005C04" w:rsidRDefault="00A80F5E" w:rsidP="00A80F5E">
                            <w:pPr>
                              <w:rPr>
                                <w:rFonts w:cstheme="minorHAnsi"/>
                                <w:noProof/>
                                <w:sz w:val="28"/>
                                <w:szCs w:val="28"/>
                                <w:bdr w:val="none" w:sz="0" w:space="0" w:color="auto" w:frame="1"/>
                              </w:rPr>
                            </w:pPr>
                            <w:r w:rsidRPr="00124A5C">
                              <w:rPr>
                                <w:rFonts w:cstheme="minorHAnsi"/>
                                <w:noProof/>
                                <w:bdr w:val="none" w:sz="0" w:space="0" w:color="auto" w:frame="1"/>
                              </w:rPr>
                              <w:fldChar w:fldCharType="begin"/>
                            </w:r>
                            <w:r w:rsidRPr="00124A5C">
                              <w:rPr>
                                <w:rFonts w:cstheme="minorHAnsi"/>
                                <w:noProof/>
                                <w:bdr w:val="none" w:sz="0" w:space="0" w:color="auto" w:frame="1"/>
                              </w:rPr>
                              <w:instrText xml:space="preserve"> SEQ Abbildung \* ARABIC </w:instrText>
                            </w:r>
                            <w:r w:rsidRPr="00124A5C">
                              <w:rPr>
                                <w:rFonts w:cstheme="minorHAnsi"/>
                                <w:noProof/>
                                <w:bdr w:val="none" w:sz="0" w:space="0" w:color="auto" w:frame="1"/>
                              </w:rPr>
                              <w:fldChar w:fldCharType="separate"/>
                            </w:r>
                            <w:r>
                              <w:rPr>
                                <w:rFonts w:cstheme="minorHAnsi"/>
                                <w:noProof/>
                                <w:bdr w:val="none" w:sz="0" w:space="0" w:color="auto" w:frame="1"/>
                              </w:rPr>
                              <w:t>1</w:t>
                            </w:r>
                            <w:r w:rsidRPr="00124A5C">
                              <w:rPr>
                                <w:rFonts w:cstheme="minorHAnsi"/>
                                <w:noProof/>
                                <w:bdr w:val="none" w:sz="0" w:space="0" w:color="auto" w:frame="1"/>
                              </w:rPr>
                              <w:fldChar w:fldCharType="end"/>
                            </w:r>
                            <w:r>
                              <w:t xml:space="preserve"> </w:t>
                            </w:r>
                            <w:r w:rsidRPr="009D25E4">
                              <w:t xml:space="preserve">Daniel </w:t>
                            </w:r>
                            <w:proofErr w:type="spellStart"/>
                            <w:r w:rsidRPr="009D25E4">
                              <w:t>Schliebner</w:t>
                            </w:r>
                            <w:proofErr w:type="spellEnd"/>
                            <w:r>
                              <w:t xml:space="preserve">: </w:t>
                            </w:r>
                            <w:proofErr w:type="spellStart"/>
                            <w:r>
                              <w:t>Cantor'sches</w:t>
                            </w:r>
                            <w:proofErr w:type="spellEnd"/>
                            <w:r>
                              <w:t xml:space="preserve"> Diagonalverfahren. Von Mengen, Unendlichkeiten und Wahnsinn, </w:t>
                            </w:r>
                            <w:proofErr w:type="spellStart"/>
                            <w:r>
                              <w:t>Pdf</w:t>
                            </w:r>
                            <w:proofErr w:type="spellEnd"/>
                            <w:r>
                              <w:t xml:space="preserve">: </w:t>
                            </w:r>
                            <w:r w:rsidRPr="00D81AD7">
                              <w:t>https://www2.informatik.hu-berlin.de/~kossahl/Uni/Ma1/Cantor.pdf</w:t>
                            </w:r>
                          </w:p>
                        </w:txbxContent>
                      </wps:txbx>
                      <wps:bodyPr rot="0" spcFirstLastPara="0" vertOverflow="overflow" horzOverflow="overflow" vert="horz" wrap="square" lIns="0" tIns="0" rIns="0" bIns="0" numCol="1" spcCol="0" rtlCol="0" fromWordArt="0" anchor="t" anchorCtr="0" forceAA="0" compatLnSpc="1">
                        <a:prstTxWarp prst="textNoShape">
                          <a:avLst/>
                        </a:prstTxWarp>
                        <a:spAutoFit/>
                      </wps:bodyPr>
                    </wps:wsp>
                  </a:graphicData>
                </a:graphic>
                <wp14:sizeRelH relativeFrom="margin">
                  <wp14:pctWidth>0</wp14:pctWidth>
                </wp14:sizeRelH>
              </wp:anchor>
            </w:drawing>
          </mc:Choice>
          <mc:Fallback>
            <w:pict>
              <v:shapetype w14:anchorId="4AFB6FA3" id="_x0000_t202" coordsize="21600,21600" o:spt="202" path="m,l,21600r21600,l21600,xe">
                <v:stroke joinstyle="miter"/>
                <v:path gradientshapeok="t" o:connecttype="rect"/>
              </v:shapetype>
              <v:shape id="Textfeld 2" o:spid="_x0000_s1026" type="#_x0000_t202" style="position:absolute;margin-left:0;margin-top:88.15pt;width:198.3pt;height:.05pt;z-index:251660288;visibility:visible;mso-wrap-style:square;mso-width-percent:0;mso-wrap-distance-left:9pt;mso-wrap-distance-top:0;mso-wrap-distance-right:9pt;mso-wrap-distance-bottom:0;mso-position-horizontal:center;mso-position-horizontal-relative:text;mso-position-vertical:absolute;mso-position-vertical-relative:text;mso-width-percent:0;mso-width-relative:margin;v-text-anchor:top" o:gfxdata="UEsDBBQABgAIAAAAIQC2gziS/gAAAOEBAAATAAAAW0NvbnRlbnRfVHlwZXNdLnhtbJSRQU7DMBBF&#10;90jcwfIWJU67QAgl6YK0S0CoHGBkTxKLZGx5TGhvj5O2G0SRWNoz/78nu9wcxkFMGNg6quQqL6RA&#10;0s5Y6ir5vt9lD1JwBDIwOMJKHpHlpr69KfdHjyxSmriSfYz+USnWPY7AufNIadK6MEJMx9ApD/oD&#10;OlTrorhX2lFEilmcO2RdNtjC5xDF9pCuTyYBB5bi6bQ4syoJ3g9WQ0ymaiLzg5KdCXlKLjvcW893&#10;SUOqXwnz5DrgnHtJTxOsQfEKIT7DmDSUCaxw7Rqn8787ZsmRM9e2VmPeBN4uqYvTtW7jvijg9N/y&#10;JsXecLq0q+WD6m8AAAD//wMAUEsDBBQABgAIAAAAIQA4/SH/1gAAAJQBAAALAAAAX3JlbHMvLnJl&#10;bHOkkMFqwzAMhu+DvYPRfXGawxijTi+j0GvpHsDYimMaW0Yy2fr2M4PBMnrbUb/Q94l/f/hMi1qR&#10;JVI2sOt6UJgd+ZiDgffL8ekFlFSbvV0oo4EbChzGx4f9GRdb25HMsYhqlCwG5lrLq9biZkxWOiqY&#10;22YiTra2kYMu1l1tQD30/bPm3wwYN0x18gb45AdQl1tp5j/sFB2T0FQ7R0nTNEV3j6o9feQzro1i&#10;OWA14Fm+Q8a1a8+Bvu/d/dMb2JY5uiPbhG/ktn4cqGU/er3pcvwCAAD//wMAUEsDBBQABgAIAAAA&#10;IQCFTAkEKwIAAF0EAAAOAAAAZHJzL2Uyb0RvYy54bWysVMGO2yAQvVfqPyDujZO0u1pZcVZpVqkq&#10;RbsrJdWeCYYYCRgKJHb69R2wnW23PVW9kGFmePi9N2Rx3xlNzsIHBbais8mUEmE51MoeK/ptv/lw&#10;R0mIzNZMgxUVvYhA75fv3y1aV4o5NKBr4QmC2FC2rqJNjK4sisAbYViYgBMWixK8YRG3/ljUnrWI&#10;bnQxn05vixZ87TxwEQJmH/oiXWZ8KQWPT1IGEYmuKH5bzKvP6yGtxXLByqNnrlF8+Az2D19hmLJ4&#10;6RXqgUVGTl79AWUU9xBAxgkHU4CUiovMAdnMpm/Y7BrmROaC4gR3lSn8P1j+eH72RNUVnVNimUGL&#10;9qKLUuiazJM6rQslNu0ctsXuM3To8pgPmEykO+lN+kU6BOuo8+WqLYIRjsn5zezu0wxLHGu3H28S&#10;RvF61PkQvwgwJAUV9Whc1pOdtyH2rWNLuimAVvVGaZ02qbDWnpwZmtw2KooB/LcubVOvhXSqB0yZ&#10;IvHreaQododuIH2A+oKcPfQzExzfKLxoy0J8Zh6HBLng4McnXKSGtqIwRJQ04H/8LZ/60TusUtLi&#10;0FU0fD8xLyjRXy26miZ0DPwYHMbAnswakOIMn5TjOcQDPuoxlB7MC76HVboFS8xyvKuicQzXsR99&#10;fE9crFa5CefQsbi1O8cT9Cjovnth3g12RHTxEcZxZOUbV/re7ItbnSJKnC1LgvYqDjrjDGfTh/eW&#10;Hsmv+9z1+q+w/AkAAP//AwBQSwMEFAAGAAgAAAAhAEbmOzLeAAAACAEAAA8AAABkcnMvZG93bnJl&#10;di54bWxMj8FOwzAQRO9I/IO1SFwQdSCRgRCnqio4wKUi9MLNjbdxIF5HttOGv8f0AsedGc2+qZaz&#10;HdgBfegdSbhZZMCQWqd76iRs35+v74GFqEirwRFK+MYAy/r8rFKldkd6w0MTO5ZKKJRKgolxLDkP&#10;rUGrwsKNSMnbO29VTKfvuPbqmMrtwG+zTHCrekofjBpxbbD9aiYrYVN8bMzVtH96XRW5f9lOa/HZ&#10;NVJeXsyrR2AR5/gXhl/8hA51Ytq5iXRgg4Q0JCb1TuTAkp0/CAFsd1IK4HXF/w+ofwAAAP//AwBQ&#10;SwECLQAUAAYACAAAACEAtoM4kv4AAADhAQAAEwAAAAAAAAAAAAAAAAAAAAAAW0NvbnRlbnRfVHlw&#10;ZXNdLnhtbFBLAQItABQABgAIAAAAIQA4/SH/1gAAAJQBAAALAAAAAAAAAAAAAAAAAC8BAABfcmVs&#10;cy8ucmVsc1BLAQItABQABgAIAAAAIQCFTAkEKwIAAF0EAAAOAAAAAAAAAAAAAAAAAC4CAABkcnMv&#10;ZTJvRG9jLnhtbFBLAQItABQABgAIAAAAIQBG5jsy3gAAAAgBAAAPAAAAAAAAAAAAAAAAAIUEAABk&#10;cnMvZG93bnJldi54bWxQSwUGAAAAAAQABADzAAAAkAUAAAAA&#10;" stroked="f">
                <v:textbox style="mso-fit-shape-to-text:t" inset="0,0,0,0">
                  <w:txbxContent>
                    <w:p w14:paraId="6D60BF4E" w14:textId="77777777" w:rsidR="00A80F5E" w:rsidRPr="00005C04" w:rsidRDefault="00A80F5E" w:rsidP="00A80F5E">
                      <w:pPr>
                        <w:rPr>
                          <w:rFonts w:cstheme="minorHAnsi"/>
                          <w:noProof/>
                          <w:sz w:val="28"/>
                          <w:szCs w:val="28"/>
                          <w:bdr w:val="none" w:sz="0" w:space="0" w:color="auto" w:frame="1"/>
                        </w:rPr>
                      </w:pPr>
                      <w:r w:rsidRPr="00124A5C">
                        <w:rPr>
                          <w:rFonts w:cstheme="minorHAnsi"/>
                          <w:noProof/>
                          <w:bdr w:val="none" w:sz="0" w:space="0" w:color="auto" w:frame="1"/>
                        </w:rPr>
                        <w:fldChar w:fldCharType="begin"/>
                      </w:r>
                      <w:r w:rsidRPr="00124A5C">
                        <w:rPr>
                          <w:rFonts w:cstheme="minorHAnsi"/>
                          <w:noProof/>
                          <w:bdr w:val="none" w:sz="0" w:space="0" w:color="auto" w:frame="1"/>
                        </w:rPr>
                        <w:instrText xml:space="preserve"> SEQ Abbildung \* ARABIC </w:instrText>
                      </w:r>
                      <w:r w:rsidRPr="00124A5C">
                        <w:rPr>
                          <w:rFonts w:cstheme="minorHAnsi"/>
                          <w:noProof/>
                          <w:bdr w:val="none" w:sz="0" w:space="0" w:color="auto" w:frame="1"/>
                        </w:rPr>
                        <w:fldChar w:fldCharType="separate"/>
                      </w:r>
                      <w:r>
                        <w:rPr>
                          <w:rFonts w:cstheme="minorHAnsi"/>
                          <w:noProof/>
                          <w:bdr w:val="none" w:sz="0" w:space="0" w:color="auto" w:frame="1"/>
                        </w:rPr>
                        <w:t>1</w:t>
                      </w:r>
                      <w:r w:rsidRPr="00124A5C">
                        <w:rPr>
                          <w:rFonts w:cstheme="minorHAnsi"/>
                          <w:noProof/>
                          <w:bdr w:val="none" w:sz="0" w:space="0" w:color="auto" w:frame="1"/>
                        </w:rPr>
                        <w:fldChar w:fldCharType="end"/>
                      </w:r>
                      <w:r>
                        <w:t xml:space="preserve"> </w:t>
                      </w:r>
                      <w:r w:rsidRPr="009D25E4">
                        <w:t xml:space="preserve">Daniel </w:t>
                      </w:r>
                      <w:proofErr w:type="spellStart"/>
                      <w:r w:rsidRPr="009D25E4">
                        <w:t>Schliebner</w:t>
                      </w:r>
                      <w:proofErr w:type="spellEnd"/>
                      <w:r>
                        <w:t xml:space="preserve">: </w:t>
                      </w:r>
                      <w:proofErr w:type="spellStart"/>
                      <w:r>
                        <w:t>Cantor'sches</w:t>
                      </w:r>
                      <w:proofErr w:type="spellEnd"/>
                      <w:r>
                        <w:t xml:space="preserve"> Diagonalverfahren. Von Mengen, Unendlichkeiten und Wahnsinn, </w:t>
                      </w:r>
                      <w:proofErr w:type="spellStart"/>
                      <w:r>
                        <w:t>Pdf</w:t>
                      </w:r>
                      <w:proofErr w:type="spellEnd"/>
                      <w:r>
                        <w:t xml:space="preserve">: </w:t>
                      </w:r>
                      <w:r w:rsidRPr="00D81AD7">
                        <w:t>https://www2.informatik.hu-berlin.de/~kossahl/Uni/Ma1/Cantor.pdf</w:t>
                      </w:r>
                    </w:p>
                  </w:txbxContent>
                </v:textbox>
                <w10:wrap type="topAndBottom"/>
              </v:shape>
            </w:pict>
          </mc:Fallback>
        </mc:AlternateContent>
      </w:r>
    </w:p>
```

**Comment by jgm** [comment]
How did you insert this caption in Word?

**Comment by Ultrahero** [comment]
Thanks,that's actually great.
-------- Ursprüngliche Nachricht --------Von: John MacFarlane <notifications@github.com> Datum: 26.11.20  20:56  (GMT+01:00) An: jgm/pandoc <pandoc@noreply.github.com> Cc: Ultrahero <bene.gerlach@t-online.de>, Author <author@noreply.github.com> Betreff: Re: [jgm/pandoc] docx to latex: images and captions are not being
  converted properly (#6893) 
I can look into the missing caption.
As for

the image itself was not included only the reference

what you need is to use the --extract-media=DIR option.  Then the image will be extracted.  Pandoc won't litter your file system with files unless you explicitly tell it where they should go.

**Comment by Ultrahero** [comment]
I am using word 365 and just left-clicked on the image, then clicked on add image caption. Then inserted my text in the text field.
-------- Ursprüngliche Nachricht --------Von: John MacFarlane <notifications@github.com> Datum: 26.11.20  21:02  (GMT+01:00) An: jgm/pandoc <pandoc@noreply.github.com> Cc: Ultrahero <bene.gerlach@t-online.de>, Author <author@noreply.github.com> Betreff: Re: [jgm/pandoc] docx to latex: images and captions are not being
  converted properly (#6893) 
How did you insert this caption in Word?

**Comment by jgm** [comment]
@jkr if this is a normal way of including captions, it would be nice to extract the content somehow.

**Comment by Yorkshire-Jack** [comment]
This would also be useful to me, as this is how I usually add captions.  If this is not supported (yet) is there a way of inserting captions which will result in them being processed correctly?

**Comment by Yorkshire-Jack** [comment]
I've had a go at righting a Lua filter to bodge together figures from the output of the docx reader:

```lua
-- Pandoc Lua Filter for fixing figures from the docx reader
-- 
-- The reader does not associate captions with images or create figures
-- This filter converts an image and the following paragraph to a figure
-- It works as long as the image and caption are each in a paragraph of
-- their own (the right click -> insert caption will be, and the image
-- usually is too) and the caption is right after the image. If an image
-- is in a paragraph on its own and has no caption, the next paragraph may
-- be used as a caption by mistake. It doesn't work if the docx+styles
-- extension is used as the caption paragraph will be wrapped in a div.


function Blocks(blocks)
  for i = #blocks-1, 1, -1 do
    if blocks[i].t == "Para" and blocks[i].content[1].t == "Image" then
      nextpara = assert(blocks[i+1], "No block to use as caption")
      assert(nextpara.t == "Para", "Candidate caption was not a Para")
      fig = pandoc.Figure(blocks[i].content, {nextpara.content})
      blocks[i] = fig
      blocks:remove(i+1)
    end
  end
  return blocks
end
```
It's a bit fragile and will only work if the docx has the right structure.  A better approach would be to look for paragraphs with the "Caption" style in word. But that information is stripped by the time the lua filter gets a look at it (unless docx+styles is used, but I don't want the other features of the styles extension).

I'm new to Lua and new to pandoc, so any feedback on the above filter will be gratefully received.

**Comment by jgm** [comment]
Interesting. With current HEAD, we get JUST the caption, no image.
But if we remove the AlternateContent element, we get the image but no caption.  (That element contains the caption, but it comes *after* the image; the image is not here contained in an AlternateContent element.)
The rules we've added to pandoc for handling ALtContent are causing pandoc to skip other things in the same run.

**Comment by jgm** [comment]
OK, I see the reason.
https://github.com/jgm/pandoc/blob/main/src/Text/Pandoc/Readers/Docx/Parse.hs#L162-L163
in this case is removing everything *but* the AlternateContent in this w:p.
Note that this is here in `unwrap` because we have the structure w:p > w:r > pict > shape > textbox > txbxcontent > p > w:r.  A run can't contain a paragraph so we have to handle this in a special way. However, our current way of handling it doesn't work in the case where a paragraph contains a textbox *and* other content; in that case the whole paragraph gets replaced by the content of the textbox, and the other stuff is ignored.
